server/src/main/java/org/elasticsearch/index/shard/IndexingStats.java

```diff
@@ -133,7 +133,12 @@ public long getDeleteCount() {
         /**
          * The total amount of time spend on executing delete operations.
          */
-        public TimeValue getDeleteTime() { return new TimeValue(deleteTimeInMillis); }
+        public TimeValue getDeleteTime() {
+            if (deleteTimeInMillis >= -1) {
```

Comment:
Wait, this shouldn't be necessary? I see tests we are passing in `-1`? Those aren't realistic though, right?

Reply:
The tests are passing in `-1` but multiple of these get added together, accumulating to larger negative numbers.

Reply:
This was the part of this PR that I felt least good about. I would be happy to change this if there's another tactic that would be better.

Reply:
Those aren't realistic though, right? Negative values can never happen in production, or am I reading it wrongly? If that's the case, this tests are fictional, and we shouldn't change production code, but change the tests to be more realistic.

Reply:
Good point, for some reason I thought these were using `-1` as a sentinel value as we do in some other places, but upon a closer look that turned out to be wrong. I've reverted the changes to the `*Stats` classes and fixed the tests instead.